heron/dsl/src/java/com/twitter/heron/dsl/KeyValue.java

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.dsl;
+
+import java.io.Serializable;
+
+/**
+ * Certain operations in dsl like join/reduce necessitate the concept of
+ * key value pairs. This file defines a generic KeyValue class. We make
+ * the KeyValue class serializable to allow it to be serialized between
+ * components.
+ */
+public class KeyValue<K, V> implements Serializable {
```

Comment:
Should this be immutable? Also maybe add a static create(K key, V value) method for connivence?

Reply:
Done